packages/glimmer-runtime/lib/compiled/opcodes/builder.ts

```diff
@@ -74,7 +77,7 @@ export abstract class BasicOpcodeBuilder extends StatementCompilationBufferProxy
     super(inner);
   }
 
-  abstract compile<E>(expr: CompilesInto<E>): E;
+  abstract compile<E>(expr: Represents<E> | E): E;
```

Comment:
Don't know much about TypeScript, but given `Represents<E> = CompilesInto<E> | E`, is it necessary to add the `| E` here too?